db/db_test2.cc

```diff
@@ -8061,6 +8062,65 @@ TEST_F(DBTest2, TableCacheMissDuringReadFromBlockCacheTier) {
   ASSERT_EQ(orig_num_file_opens, TestGetTickerCount(options, NO_FILE_OPENS));
 }
 
+TEST_F(DBTest2, GetFileChecksumsFromCurrentManifest_CRC32) {
+  const int kNumL0Files = 5;
+
+  Options opts = GetDefaultOptions();
+  opts.create_if_missing = true;
+  opts.file_checksum_gen_factory = GetFileChecksumGenCrc32cFactory();
+  opts.level0_file_num_compaction_trigger = kNumL0Files;
+
+  DB* db = nullptr;
+  std::string dbname = test::PerThreadDBPath("file_chksum");
+  ASSERT_OK(DB::Open(opts, dbname, &db));
+
+  WriteOptions wopts;
+  FlushOptions fopts;
+  fopts.wait = true;
+  Random rnd(test::RandomSeed());
+  for (int i = 0; i < kNumL0Files; i++) {
+    char buf[16];
+    snprintf(buf, sizeof(buf), "%08d", i);
+    std::string v = rnd.RandomString(100);
+    ASSERT_OK(db->Put(wopts, buf, v));
+    ASSERT_OK(db->Flush(fopts));
+  }
+
+  std::vector<LiveFileMetaData> live_files;
+  db->GetLiveFilesMetaData(&live_files);
+
+  ASSERT_OK(db->Close());
+  delete db;
+  db = nullptr;
+
+  std::unique_ptr<FileChecksumList> checksum_list(NewFileChecksumList());
+  auto read_only_fs = new ReadOnlyFileSystem(env_->GetFileSystem());
```

Comment:
We sometimes use raw pointers with things like DB because legacy public APIs are difficult to change. (Note that DBTest2.PresetCompressionDictLocality does not!) In general, if you can use modern idioms like make_unique, please do your part to normalize less bug-prone idioms.